pkg/ebpf/processor_funcs.go

```diff
@@ -21,112 +18,7 @@ import (
 	"github.com/aquasecurity/tracee/types/trace"
 )
 
-var (
-	// initializing kernelReadFileTypes once at init.
-	kernelReadFileTypes map[int32]trace.KernelReadType
-	// exec hash might add capabilities to base ring
-	onceExecHash sync.Once
-)
-
-func init() {
-	initKernelReadFileTypes()
-}
-
-func (t *Tracee) processLostEvents() {
-	logger.Debugw("Starting processLostEvents goroutine")
-	defer logger.Debugw("Stopped processLostEvents goroutine")
-
-	// Since this is an end-stage goroutine, it should be terminated when:
-	// - lostEvChannel is closed, or finally when;
-	// - internal done channel is closed (not ctx).
-	for {
-		select {
-		case lost, ok := <-t.lostEvChannel:
-			if !ok {
-				return // lostEvChannel is closed, lost is zero value
-			}
-
-			if err := t.stats.LostEvCount.Increment(lost); err != nil {
-				logger.Errorw("Incrementing lost event count", "error", err)
-			}
-			logger.Warnw(fmt.Sprintf("Lost %d events", lost))
-
-		// internal done channel is closed when Tracee is stopped via Tracee.Close()
-		case <-t.done:
-			return
-		}
-	}
-}
-
-const (
-	IterateShared int = iota
-	Iterate
-)
-
-func (t *Tracee) processEvent(event *trace.Event) []error {
-	eventId := events.ID(event.EventID)
-	processors := t.eventProcessor[eventId]
-	errs := []error{}
-	for _, procFunc := range processors {
-		err := procFunc(event)
-		if err != nil {
-			logger.Errorw("Error processing event", "event", event.EventName, "error", err)
-			errs = append(errs, err)
-		}
-	}
-	return errs
-}
-
-// RegisterEventProcessor registers a pipeline processing handler for an event
-func (t *Tracee) RegisterEventProcessor(id events.ID, proc func(evt *trace.Event) error) error {
-	if t.eventProcessor == nil {
-		return errfmt.Errorf("tracee not initialized yet")
-	}
-	if t.eventProcessor[id] == nil {
-		t.eventProcessor[id] = make([]func(evt *trace.Event) error, 0)
-	}
-	t.eventProcessor[id] = append(t.eventProcessor[id], proc)
-	return nil
-}
-
-// registerEventProcessors registers tracee's internal default event processors
-func (t *Tracee) registerEventProcessors() {
-	if t.eventProcessor == nil {
-		t.eventProcessor = make(map[events.ID][]func(evt *trace.Event) error)
-	}
-
-	// no need to error check since we know the error is initialization related
-	t.RegisterEventProcessor(events.VfsWrite, t.processWriteEvent)
-	t.RegisterEventProcessor(events.VfsWritev, t.processWriteEvent)
-	t.RegisterEventProcessor(events.KernelWrite, t.processWriteEvent)
-	t.RegisterEventProcessor(events.SchedProcessExec, t.processSchedProcessExec)
-	t.RegisterEventProcessor(events.SchedProcessFork, t.processSchedProcessFork)
```

Comment:
No internal rule using "sched_process_fork" event was accessing "start_time" in any means I could find. All of them access pid/tid or are helpers for binary data LRU or the process tree (still being worked).

Check it yourself: 

```
$ for file in $(find . -name *.go -exec grep -il sched_process_for
k {} \;); do grep -Fil time $file; done
```

Reply:
The idea here was not doing it for a specific signature.
The idea was that getting time not since epoch is not useful except for low-level code.
The user expect to get all times in unix nano seconds (aka since epoch), as it is with event timestamp and other times.
Actually, I thing we even should create a type for time, which automatically upon parsing from the kernel will be changed to unix time so all times will always match what the user expects.

To conclude, I am againts removing it and for even expanding it to all the time arguments we currently have.

Reply:
Sounds good to me (about the time type). About Fork, if all you want is for the "event" to contain the NS since epoch, to normalize with all other times, then it sounds good to me and I can keep it.

Reply:
![image](https://github.com/aquasecurity/tracee/assets/7395852/1870f975-d8d4-4b95-932f-4234fd770d4d)

I thought that checking if user asked for relative time or not was better. This way all timings are either relative or not:

![image](https://github.com/aquasecurity/tracee/assets/7395852/2b865e9a-6cc0-4de6-a94e-07766b8b4cc0)

Bad thing is that all events using "timing arguments" should declare themselves and convert their args like the fork event example.